service/routes.py

```diff
@@ -46,123 +38,318 @@ def index():
+class PromotionResource(Resource):
     """
-    Read details of specific promotion id
+    PromotionResource class
+
+    Allows the manipulation of a single Promotion
+    GET /promotion/{promotion_id} - Get details of a Promotion with the promotion_id
+    PUT /promotion/{promotion_id} - Update a Promotion with the promotion_id
+    DELETE /promotion/{promotion_id} - Delete a Promotion with the promotion_id
     """
-    app.logger.info(
-        "Request to Retrieve a promotion with promotion id [%s]", promotion_id
-    )
-    promotion = Promotion.find(promotion_id)
-    if not promotion:
-        abort(
-            status.HTTP_404_NOT_FOUND,
-            f"Promotion with id '{promotion_id}' was not found.",
-        )
-    app.logger.info("Returning promotion: %s", promotion.promotion_name)
-    return jsonify(promotion.serialize()), status.HTTP_200_OK
 
+    @api.doc("update_promotion")
+    @api.response(400, "The posted Promotion data was not valid")
+    @api.response(404, "Promotion not found")
+    @api.marshal_with(promotion_model)
+    @api.expect(update_model)
+    def put(self, promotion_id):
+        """
+        Updates a promotion with the specified promotion_id
+        Returns the updated promotion after the update has been applied"""
+        app.logger.info(f"Got request to update Promotion with id: {promotion_id}")
+        promotion = Promotion.find(promotion_id)
+        if not promotion:
+
+            abort_with_error(
+                status.HTTP_404_NOT_FOUND, f"Promotion with id: {promotion_id} not found"
+            )
+        request_json = request.get_json()
+        promotion = promotion.deserialize(request_json)
+        promotion.update()
+        return promotion.serialize()
+
+    @api.doc("get_promotion")
+    @api.response(404, "Promotion not found")
+    @api.marshal_with(promotion_model)
+    def get(self, promotion_id):
+        """
+        Read details of specific promotion id
+        Returns promotion details of a single promotion with the specified promotion id
+        """
+        app.logger.info(
+            "Request to Retrieve a promotion with promotion id [%s]", promotion_id
+        )
+        promotion = Promotion.find(promotion_id)
+        if not promotion:
+            abort(
+                status.HTTP_404_NOT_FOUND,
+                f"Promotion with id '{promotion_id}' was not found.",
+            )
+        app.logger.info("Returning promotion: %s", promotion.promotion_name)
+        return promotion.serialize(), status.HTTP_200_OK
+
+    @api.doc("delete_promotion")
+    @api.response(204, "Promotion deleted")
+    @api.response(404, "Promotion not found")
+    def delete(self, promotion_id):
+        """
+        Deletes a Promotion with promotion_id
+        """
+        app.logger.info(f"Got request to delete Promotion with id: {promotion_id}")
+        promotion = Promotion.find(promotion_id)
+        if promotion:
+            promotion.delete()
+
```

Comment:
Do we also return HTTP_204_NO_CONTENT when the ID does not exist? I reviewed the test cases, and it seems there is some divergence.
![image](https://github.com/user-attachments/assets/54bb3697-1fd2-4ea6-908e-15d2d414afde)

Reply:
It should be a 204
![image](https://github.com/user-attachments/assets/03a16bf2-eba3-48eb-b52c-ba31a4748a61)

Let me change the spec string. Good catch!

Reply:
Thanks for clarifying!

Reply:
Got it thanks!